azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java

```diff
@@ -131,335 +63,47 @@
     @Parameter(property = "deployment")
     protected AppDeploymentMavenConfig deployment;
 
-    @Getter
-    @Parameter(defaultValue = "${project}", readonly = true, required = true)
-    protected MavenProject project;
-
-    @Getter
-    @Parameter(defaultValue = "${session}", readonly = true, required = true)
-    protected MavenSession session;
-
-    @Getter
-    @Parameter(defaultValue = "${project.build.directory}", readonly = true, required = true)
-    protected File buildDirectory;
-
-    @Getter
-    @Parameter(defaultValue = "${plugin}", readonly = true, required = true)
-    protected PluginDescriptor plugin;
-
-    @Getter
-    protected Map<String, String> telemetries;
-
-    @Component
-    protected SettingsDecrypter settingsDecrypter;
-
     @Parameter(defaultValue = "${settings}", readonly = true)
     protected Settings settings;
 
-    protected Long timeStart;
-
-    @Parameter(property = "authType")
-    protected String authType;
-
-    /**
-     * Use a HTTP proxy host for the Azure Auth Client
-     */
-    @Parameter(property = "httpProxyHost")
-    protected String httpProxyHost;
-
-    /**
-     * Use a HTTP proxy port for the Azure Auth Client
-     */
-    @Parameter(property = "httpProxyPort")
-    protected String httpProxyPort;
-
+    @SneakyThrows
     @Override
-    public void execute() throws MojoFailureException, MojoExecutionException {
-        try {
-            initExecution();
-            doExecute();
-            handleSuccess();
-        } catch (Exception e) {
-            if (e instanceof AzureToolkitAuthenticationException) {
-                throw new MojoExecutionException(String.format("Cannot authenticate due to error: %s", e.getMessage()), e);
-            }
-            handleException(e);
-            throw new MojoFailureException(e.getMessage(), e);
-        }
-    }
-
-    protected void initExecution() throws MavenDecryptException, AzureExecutionException, LoginFailureException {
-        AzureMessager.setDefaultMessager(new MavenAzureMessager());
-        // init proxy manager
-        ProxyUtils.initProxy(Optional.ofNullable(this.session).map(MavenSession::getRequest).orElse(null));
-        // Init telemetries
-        initTelemetry();
-        telemetries.put(PROXY, String.valueOf(ProxyManager.getInstance().getProxy() != null));
-        Azure.az().config().setLogLevel(HttpLogDetailLevel.NONE.name());
-        Azure.az().config().setUserAgent(getUserAgent());
-        trackMojoExecution(MojoStatus.Start);
+    protected void beforeMojoExecution() {
+        super.beforeMojoExecution();
```

Comment:
Whether we do authentication in each goal of spring plugin? Besides, do we need to do some static validation first before authentication.